### DISCUSSION TITLE
Video Preview thumbnail

### DISCUSSION THREAD

**Discussion by electricpollution** [photoprism/photoprism#1235] [Ideas]
Many of my videos have the first several frames either to be the same, or black. Is there any way to tell it what frame or how many seconds in to generate the preview image. Something with ffmpeg?  
`-ss 00:00:10.000 -vframes 1`

**Comment by lastzero** [comment]
I guess we could use https://github.com/dirkvdb/ffmpegthumbnailer, which should already be installed in our Docker image. Just didn't have time to look into this. Right now, we use one of the very first frames. FFmpeg otherwise returns an error for (very) short videos, so we'd be stuck in a try & error loop.

**Reply by electricpollution** [reply]
Perfect thanks. I am happy to throw some sponsorship for this. Maybe default to first frame for thumbnail, and have some sort of parameter to customize it, e.g. x amount of seconds in.

Photoprism is fantastic. I use it house a media collection of 800K images and videos

**Reply by graciousgrey** [reply]
@electricisfun could you open an issue (https://github.com/photoprism/photoprism/issues) for this. This way we can add it as idea to our roadmap :)

**Reply by StephenBrown2** [reply]
`-t: time to seek to (percentage or absolute time hh:mm:ss) (default: 10)` seems perfect!

**Reply by lastzero** [reply]
Won't this error if the video is just 3 seconds long? Percentage might work...

**Reply by StephenBrown2** [reply]
Yes, I meant to point to the fact that it can use percentage, thus the default of 10% should work fine for anything.

**Reply by G2G2G2G** [reply]
Looks like there is a "pixel shift sensing flag" within ffmpeg to catch action frames within videos. The percentages are a good method that I didn't know about, however extracting an image has no better solution than what I posted: https://github.com/photoprism/photoprism/issues/1241#issuecomment-853613941
